File: src/interfaces/user-settings.ts
import { prop, Typegoose, ModelType, InstanceType } from 'typegoose';
import IAccountSettings from './account-settings';

export default class IUserSettings extends IAccountSettings {

	@prop({ required: true })
	is_full_name_visible: boolean;

	@prop({ required: true })
	blocked_users: Array<string>;

}

File: src/interfaces-internal/user-settings.go
package interfaces_internal

type IUserSettings struct {
	AllowMessagesFromUnknown bool   `bson:"allow_messages_from_unknown"`
	EmailNotifications       bool   `bson:"email_notifications"`
	// user specific settings
	IsFullNameVisible bool     `bson:"is_full_name_visible"`
	BlockedUsers      []string `bson:"blocked_users"`
}
